### PR TITLE
Add parallelstore module and support for rocky 8, ubuntu 22.04 and     debian 12

### PR DESCRIPTION
**Pull request by harshthakkar01**
This PR adds a skeleton code for parallelstore module. 

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: modules/file-system/parallelstore/variables.tf [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+variable "deployment_name" {
+  description = "Name of the HPC deployment."
+  type        = string
+}
+
+variable "instance_id" {
+  description = "Name of parallelstore instance."
+  type        = string
+  default     = "parallelstore-instance"
```
Comment (mr0re1) [inline modules/file-system/parallelstore/variables.tf:30]: why provide default? wouldn't it lead to collisions ?

File: modules/file-system/parallelstore/variables.tf [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+variable "labels" {
+  description = "Labels to add to parallel store instance."
+  type        = map(string)
+}
+
+variable "local_mount" {
+  description = "The mount point where the contents of the device may be accessed after mounting."
+  type        = string
+  default     = "/tmp/parallelstore"
```
Comment (mr0re1) [inline modules/file-system/parallelstore/variables.tf:57]: why `/tmp/*`, and not just `/parallelstore` ?

File: modules/file-system/parallelstore/install_and_mount_parallelstore.sh [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	# 3) Increase open file limit, for Ubuntu 22.04 only.
+	# This is necessary for dfuse and for the interception library.
+	if grep -q "ID=ubuntu" /etc/os-release && lsb_release -rs | grep -q "22\.04"; then
+		tee -a /etc/security/limits.conf <<EOF
+* soft nofile 131072
+* hard nofile 131072
+EOF
+	fi
+
+	reboot
```
Comment (mr0re1) [inline modules/file-system/parallelstore/install_and_mount_parallelstore.sh:94]: I can't wrap my head around this `reboot`. Let's discuss offline.

File: modules/file-system/parallelstore/variables.tf [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+variable "capacity_gib" {
+  description = "Storage size of the parallelstore instance in GB."
+  type        = number
+  default     = 12000
+}
+
+variable "labels" {
+  description = "Labels to add to parallel store instance."
+  type        = map(string)
```
Comment (mr0re1) [inline modules/file-system/parallelstore/variables.tf:51]: Please add `default = {}` to improve usability outside of toolkit )

Comment (harshthakkar01) [reply modules/file-system/parallelstore/variables.tf:51]: Its not updated https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/modules/file-system/filestore/variables.tf#L103 

Will we update this everywhere ?

File: modules/file-system/parallelstore/variables.tf (hunk 2 above)
Comment (harshthakkar01) [reply modules/file-system/parallelstore/variables.tf]: This was for POC local testing purpose. Updated.

File: modules/file-system/parallelstore/variables.tf (hunk 1 above)
Comment (harshthakkar01) [reply modules/file-system/parallelstore/variables.tf]: This was for POC local testing purpose. Updated.

File: modules/file-system/parallelstore/install_and_mount_parallelstore.sh (hunk 3 above)
Comment (harshthakkar01) [reply modules/file-system/parallelstore/install_and_mount_parallelstore.sh]: Its a startup script to install drivers for ubuntu and debian. I will ping you offline. For now, this PR focuses on rocky.

File: modules/file-system/parallelstore/variables.tf (hunk 4 above)
Comment (mr0re1) [reply modules/file-system/parallelstore/variables.tf:51]: It's low priority, no need to update it everywhere.

**Review by mr0re1 — APPROVED** [review]

File: modules/file-system/parallelstore/variables.tf (hunk 4 above)
Comment (harshthakkar01) [reply modules/file-system/parallelstore/variables.tf:46]: Done.

**Review by mr0re1 — APPROVED** [review]

**Pull request merged**